### PR TITLE
feat(Server):support Verbatim strings resp type, using it for CLIENT LIST and INFO commands

### PR DESCRIPTION
**Pull request by theyueli**
fixes https://github.com/dragonflydb/dragonfly/issues/2242 https://github.com/dragonflydb/dragonfly/issues/2253

Reference:
The definition of Redis verbatim strings: https://redis.io/docs/reference/protocol-spec/#verbatim-strings

"For example, the Redis command [INFO](https://redis.io/commands/info) outputs a report that includes newlines. When using RESP3, redis-cli displays it correctly because it is sent as a Verbatim String reply (with its three bytes being "txt"). When using RESP2, however, the redis-cli is hard-coded to look for the [INFO](https://redis.io/commands/info) command to ensure its correct display to the user."

### CODE REVIEW

File: src/facade/reply_builder.cc [hunk 1]
```diff
@@ -328,6 +328,32 @@ void RedisReplyBuilder::SendBulkString(std::string_view str) {
...
+  std::string_view format_str;
+  if (format == VerbatimFormat::TXT)
+    format_str = "txt:";
+  else if (format == VerbatimFormat::MARKDOWN)
+    format_str = "mkd:";
+  else {
+    DVLOG(1) << "Unknown verbatim reply format: " << format;
+    return;
+  }
+  iovec v[4] = {IoVec(lenpref), IoVec(format_str), IoVec(str), IoVec(kCRLF)};
```
Comment (romange) [inline src/facade/reply_builder.cc:353]: there is a small preference to reduce number of vectors (it's more efficient).
Therefore, I suggest to increase `tmp` buffer size to accommodate the format prefix as well.

Comment (theyueli) [reply src/facade/reply_builder.cc]: ok, done.

**Review by chakaz — APPROVED** [review]

File: src/facade/reply_builder.cc [hunk 2]
```diff
@@ -328,6 +328,31 @@ void RedisReplyBuilder::SendBulkString(std::string_view str) {
...
+  char* next = absl::numbers_internal::FastIntToBuffer(uint32_t(str.size() + 4), tmp + 1);
+  *next++ = '\r';
+  *next++ = '\n';
+
+  if (format == VerbatimFormat::TXT)
+    strcpy(next, "txt:");
+  else if (format == VerbatimFormat::MARKDOWN)
+    strcpy(next, "mkd:");
+  else {
+    DVLOG(1) << "Unknown verbatim reply format: " << format;
```
[the remark this answers is not in the document]
Comment (theyueli) [reply src/facade/reply_builder.cc:347]: fixed.

File: src/facade/reply_builder.cc [hunk 3]
```diff
@@ -328,6 +328,31 @@ void RedisReplyBuilder::SendBulkString(std::string_view str) {
...
+  // + 4 because format is three byte, and need to be followed by a ":"
+  char* next = absl::numbers_internal::FastIntToBuffer(uint32_t(str.size() + 4), tmp + 1);
+  *next++ = '\r';
+  *next++ = '\n';
+
+  if (format == VerbatimFormat::TXT)
+    strcpy(next, "txt:");
+  else if (format == VerbatimFormat::MARKDOWN)
+    strcpy(next, "mkd:");
+  else {
```
[the remark this answers is not in the document]
Comment (theyueli) [reply src/facade/reply_builder.cc:346]: removed the clause due to the use of DCHECK

File: src/facade/reply_builder.cc [hunk 4]
```diff
@@ -328,6 +328,31 @@ void RedisReplyBuilder::SendBulkString(std::string_view str) {
   return Send(v, ABSL_ARRAYSIZE(v));
 }
 
+void RedisReplyBuilder::SendVerbatimString(std::string_view str, VerbatimFormat format) {
+  if (!is_resp3_)
+    return SendBulkString(str);
+
+  char tmp[absl::numbers_internal::kFastToBufferSize + 7];
```
[the remark this answers is not in the document]
Comment (theyueli) [reply src/facade/reply_builder.cc:335]: i will keep the raw array to be consistent with other places...

**Pull request merged**